model/src/plugin_handler.cpp

```diff
@@ -773,16 +773,31 @@ bool PluginHandler::archive_check(int r, const char* msg, struct archive* a) {
   return r >= ARCHIVE_WARN;
 }
 
+/*
+ * Helper function to extract a tarball into platform-specific user directories.
+ *
+ *   @param path: Path to tarball.
+ *   @param filelist: On return contains a list of files installed.
+ *   @param metadata_path: If non-empty, location where to extract plugin metadata.xml file.
+ *   @param only_metadata: If true don't install any files, just extract
+ *                         the metadata.xml file.
+ *   @return true on success.
+ *           false if metadata_path != "" and tarball does not contain metadata.xml file.
+ */
 bool PluginHandler::explodeTarball(struct archive* src, struct archive* dest,
                                    std::string& filelist,
                                    const std::string& metadata_path,
                                    bool only_metadata) {
   struct archive_entry* entry = 0;
   pathmap_t pathmap = getInstallPaths();
+  bool status = metadata_path == "";
   while (true) {
     int r = archive_read_next_header(src, &entry);
     if (r == ARCHIVE_EOF) {
-      return true;
+      if (metadata_path != "" && !status) {
+        wxLogWarning("Plugin tarball does not contain metadata.xml");
```

Comment:
wxLogWarning is for rather serious matters involving internal problems. This is just some bad external data and should be wxLogMessage

Reply:
You are right. I now realize the intent was probably to display an error message when the tarball does not contain the metadata.xml file, rather than logging a message. I.e. if user tries to import/enable a plugin and plugin is missing the `metadata.xml` file, a popup should be displayed with the error message to help the user understand why the plugin is not loaded.

Reply:
> wxLogWarning is for rather serious matters involving internal problems.

This `plugin_handler.cpp` file has 20 calls to `wxLogWarning`, all related to the tool being unable to install a plugin.